built-ins/authoring-built-ins.md

```diff
@@ -0,0 +1,132 @@
+
+After you've created the version directory. It's time to add the templates/artifacts json to the directory. If your template spec contains a template that does not leverage template spec artifacts (ie: relative paths), this process is extremely simple... Just copy your Azure Resource Manager template to:
+
+* /*{templateSpecId}*/versions/*{versionName}*/**mainTemplate.json**
+
+**If your template spec contains artifacts** you'll need to export the template spec using [Az Powershell](https://docs.microsoft.com/en-us/powershell/azure/new-azureps-module-az) cmdlets. That process is as follows:
+
+* Use `Set-AzContext` to set the subscription where your template spec is located to the active context
+* Run `Export-AzTemplateSpec -ResourceGroupName {rgName} -Name {templateSpecId} -Version {versionName} -OutputFolder {theRootBuiltInsDirectory}/{templateSpecId}/versions/{versionName}`
+* Navigate to the output folder and rename the .json file created there (it'll be named with the format "*{templateSpecId}.{versionName}.json*") to **mainTemplate.json**
+
+### Step 4: Add metadata for your template spec
+
+Each built-in template spec requires additional metadata for the overall template spec, as well as each individual version. We'll start with the overall template spec metadata....
+
+#### Metadata for overall template spec:
+Create a file in your template spec directory with the name 'metadata.json' (eg: * /*{templateSpecId}*/metadata.json) and populate it with the following contents to start things off:
+
+    {
+        "displayName" : "User Friendly Built-in Name",
+        "description": "Very short description of the built-in's purpose",
+        "documentationUrl": "https://www.microsoft.com/myDocumentation",
+        "category": "General",
+    }
+
+Replace the example values with values relevant to your built-in. The **required** properties are as follows:
+
+* **displayName**: This will be the user friendly (it can contain spaces) name displayed to customers in the Azure Portal (*currently the built-in UI is pending implementation*). It should be as short as possible and follow normal casing.
+* **description**: Very short description of the built-in's purpose. For example: "Deploys a Windows VM."
+* **category**: The relevant category for your built-in. Currently the following categories are supported (they will be expanded in the near future, but additional requests can be raised under our github issue tracking):
+  * *"General"* - Common catch-all category for general built-ins that do not fit any other available category
+  * *"Quickstart"* - Provides a starting point for a common deployment scenario (deploying a VM for example)
```

Comment:
Do we really want quickstarts to be published as built-ins? They are typically meant to be modified before using (although some users do use them as-is in production).

Reply:
I think there's value in having some common deployment scenarios that can be used as "example" template specs (especially when we have some with artifacts). But also in this case I see them more as built-ins that can be used to deploy a usually complicated resource in a streamlined (simple/quick) fashion. I definitely don't foresee us hitting parity with existing ARM template quickstarts. 

On a side note however, I do want to add a "clone/duplicate" button in the portal for our built-ins UI (which would allow users to modify built-ins in their own RGs)

Reply:
I think we should be careful with the word Quickstart since it has prior meaning. 

@stuartko -- based on how you describe the types of items you would want in this category, it sounds like it aligns more with what we've been describing as "modules", which are more generally useful without modification -- has the right number of params exposed, useful outputs, etc. That being said, I'd expect that all of the built-ins meet this requirement though, otherwise, I'm not sure it would be useful as a built-in, so I'm not sure it would help to call that out in a category.

FWIW, I was imagining categories like "Compute", "Network", "Storage", "App", etc.

What do we think about putting everything in General to start and as more as needed?

Reply:
@alex-frankel : I'm okay with putting everything in General to start with, as we can always shift the categories later being it's currently only planned to be used as a UI construct. I'll update the PR/metadata with that change. 

Regarding the possible categories, I think we want to be careful of selecting categories that may be seen as being 1 to 1 mappings with RPs (perhaps we can auto add a tag for each RP used in built-ins in the future). For example instead of Network we may want to use "Networking". We can have discussions around this as new built-ins are proposed and as the ecosystem gets ready for prime time.

Reply:
Sounds good to me